Tests/Public/Core/DatabaseQueue/DatabaseQueueTests.swift

```diff
@@ -28,4 +28,154 @@ class DatabaseQueueTests: GRDBTestCase {
+                    return (string1 as NSString).localizedStandardCompare(string2)
+                }
+                dbQueue.addCollation(collation)
+                try dbQueue.inDatabase { db in
+                    try db.execute("CREATE TABLE files (name TEXT COLLATE TEST_COLLATION_FOO)")
+                }
+                dbQueue.removeCollation(collation)
+                do {
+                    try dbQueue.inDatabase { db in
+                        try db.execute("CREATE TABLE files_fail (name TEXT COLLATE TEST_COLLATION_FOO)")
+                        XCTFail("Expected Error")
+                    }
+                    XCTFail("Expected Error")
+                }
+                catch let error as DatabaseError {
+                    // expected error
+                    XCTAssertEqual(error.code, 1) // SQLITE_ERROR
+                    XCTAssertEqual(error.message!.lowercaseString, "no such collation sequence: test_collation_foo") // lowercaseString: accept multiple SQLite version
+                    XCTAssertEqual(error.sql!, "CREATE TABLE files_fail (name TEXT COLLATE TEST_COLLATION_FOO)")
+                    XCTAssertEqual(error.description.lowercaseString, "sqlite error 1 with statement `create table files_fail (name text collate test_collation_foo)`: no such collation sequence: test_collation_foo")
+                }
+            }
+        }
+    }
+    
+    func testConfigurationSetGet() {
+        assertNoError {
+            do {
+                let inputConfiguration = dbConfiguration
+                let dbQueue = try makeDatabaseQueue()
+                var queueConfiguration = dbQueue.configuration
+                
+                // Test values of the Configurations for equality
+                // NOTE: some of the values can't be tested for equality (ex. closures)
+                
+                XCTAssertEqual(inputConfiguration.foreignKeysEnabled, queueConfiguration.foreignKeysEnabled)
+                XCTAssertEqual(inputConfiguration.readonly, queueConfiguration.readonly)
+                // can only test whether both have trace set
+                XCTAssertTrue(compareOptional(inputConfiguration.trace, with: queueConfiguration.trace))
+                #if SQLITE_HAS_CODEC
+                    XCTAssertEqual(inputConfiguration.passphrase, queueConfiguration.passphrase)
+                #endif
+                XCTAssertTrue(compare(fileAttributes: inputConfiguration.fileAttributes, with: queueConfiguration.fileAttributes))
+                XCTAssertEqual(inputConfiguration.defaultTransactionKind, queueConfiguration.defaultTransactionKind)
+                XCTAssertTrue(compare(busyMode: inputConfiguration.busyMode, with: queueConfiguration.busyMode))
+                XCTAssertEqual(inputConfiguration.maximumReaderCount, queueConfiguration.maximumReaderCount)
+                
+                //threadingMode
+                //SQLiteConnectionDidOpen
```

Comment:
SQLiteConnectionDidOpen, SQLiteConnectionWillClose, and SQLiteConnectionDidClose are not public.